app/models/payload.rb

```diff
@@ -0,0 +1,56 @@
+require 'time'
+require './app/models/parser'
+require 'user_agent_parser'
+
+class Payload < ActiveRecord::Base
+  include Parser
+
+  validates :requested_at, :responded_in, :url_id, :referred_by_id,
+  :request_type_id, :event_name_id, :u_agent_id, :resolution_id, :ip_id,
+  :client_id, presence: true
+
+  belongs_to :url
+  belongs_to :referred_by
+  belongs_to :request_type
+  belongs_to :event_name
+  belongs_to :u_agent
+  belongs_to :resolution
+  belongs_to :ip
+  belongs_to :client
+
+  def self.average_response_time
+    Payload.average(:responded_in)
```

Comment:
since we're in a class method on `Payload`, we don't need to explicitly call `average` on `Payload` here... we could just have:
```ruby
average(:responded_in)
```